binsync/ui/panel_tabs/util_panel.py

```diff
@@ -340,26 +342,43 @@ def run(self):
         parsed = urllib.parse.urlparse(self.server_url)
         if parsed.netloc != f"{host}:{port}":
             l.error("HOST AND PORT COMBINATION IS NOT VALID: NETLOC %s BUT HOST %s AND PORT %s",parsed.netloc,parsed.host,parsed.port)
-        l.info(requests.get(self.server_url+"/connect").text)
-        self.connected = True
-        while self.connected:
-            time.sleep(1)
-        l.info(requests.get(self.server_url+"/disconnect").text)
+        self.manage_connections()
         self.finished.emit()
-        
-        
+
+    def manage_connections(self):
+        sess = requests.Session()
+        try:
+            l.info(sess.get(self.server_url+"/connect").text)
+            self.connected = True
+            while self.connected:
```

Comment:
Instead of creating yur own checking loop, you should use the callback system we have built in:
https://github.com/binsync/libbs/blob/c03cd7974f11fb87cd1de79d7d1d65aa068c4459/examples/change_watcher_plugin/bs_change_watcher/__init__.py#L35

When you connect, add a function to this dictionary for type `Context`:
https://github.com/binsync/libbs/blob/c03cd7974f11fb87cd1de79d7d1d65aa068c4459/libbs/api/decompiler_interface.py#L99

Each type the `Context` is changed, you will get an alert, then do what you need with that alert.